HPXMLtoOpenStudio/resources/EPvalidator.xml

```diff
@@ -209,29 +230,41 @@
       <sch:assert role='ERROR' test='count(h:InteriorAdjacentTo[text()="living space" or text()="attic - vented" or text()="attic - unvented" or text()="basement - conditioned" or text()="basement - unconditioned" or text()="crawlspace - vented" or text()="crawlspace - unvented" or text()="garage"]) = 1'>Expected 1 element(s) for xpath: InteriorAdjacentTo[text()="living space" or text()="attic - vented" or text()="attic - unvented" or text()="basement - conditioned" or text()="basement - unconditioned" or text()="crawlspace - vented" or text()="crawlspace - unvented" or text()="garage"]</sch:assert>
       <sch:assert role='ERROR' test='count(h:Area) = 1'>Expected 1 element(s) for xpath: Area</sch:assert>
       <sch:assert role='ERROR' test='count(h:Azimuth) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Azimuth</sch:assert>
-      <sch:assert role='ERROR' test='not(h:Siding) or count(h:Siding[text()="wood siding" or text()="vinyl siding" or text()="stucco" or text()="fiber cement siding" or text()="brick veneer" or text()="aluminum siding"]) = 1'>Expected 1 element(s) for xpath: [not(Siding)] | Siding[text()="wood siding" or text()="vinyl siding" or text()="stucco" or text()="fiber cement siding" or text()="brick veneer" or text()="aluminum siding"]</sch:assert>
+      <sch:assert role='ERROR' test='count(h:Siding) &lt;= 1'>Expected 0 or 1 element(s) for xpath: Siding</sch:assert> <!-- See [RimJoistSiding] -->
       <sch:assert role='ERROR' test='count(h:SolarAbsorptance) + count(h:Color) &gt;= 1'>Expected 1 or more element(s) for xpath: SolarAbsorptance | Color</sch:assert>
       <sch:assert role='ERROR' test='count(h:Emittance) = 1'>Expected 1 element(s) for xpath: Emittance</sch:assert>
       <sch:assert role='ERROR' test='count(h:Insulation/h:SystemIdentifier) = 1'>Expected 1 element(s) for xpath: Insulation/SystemIdentifier</sch:assert>
       <sch:assert role='ERROR' test='count(h:Insulation/h:AssemblyEffectiveRValue) = 1'>Expected 1 element(s) for xpath: Insulation/AssemblyEffectiveRValue</sch:assert>
     </sch:rule>
   </sch:pattern>
 
+  <sch:pattern name='[RimJoistSiding]'>
+    <sch:rule context='/h:HPXML/h:Building/h:BuildingDetails/h:Enclosure/h:RimJoists/h:RimJoist/h:Siding'>
+      <sch:assert role='ERROR' test='contains("_wood siding_ _vinyl siding_ _stucco_ _fiber cement siding_ _brick veneer_ _aluminum siding_", concat("_", text(), "_"))'>Expected "text()" for xpath: "wood siding", "vinyl siding", "stucco", "fiber cement siding", "brick veneer", or "aluminum siding"</sch:assert>
+    </sch:rule>
+  </sch:pattern>
```

Comment:
Should we combine [WallSiding] and [RimJoistSiding] into one pattern, as you did for appliances' location?

Reply:
I agree to go in the direction of reducing the amount of code by combining these kinds of elements. I'll search for these elements and put them together.

Reply:
I've been thinking about this. I like the wildcards approach but the more I think about it, the more I lean towards option B you described in the next comment (i.e. having a separate rule for each appliance within a single pattern). I think providing clear warning messages is more important than simplifying the code. Plus, this should also be more clear for developers.